Shape_detection/include/CGAL/Shape_detection/Efficient_RANSAC/Efficient_RANSAC.h

```diff
@@ -580,8 +583,13 @@ class Efficient_RANSAC {
                    m_shape_index,
                    m_required_samples);
 
-                if (callback && !callback(num_invalid / double(m_num_total_points)))
+                if (callback && !callback(num_invalid / double(m_num_total_points))) {
+                  clear_octrees();
+                  clear_shape_factories();
+                  clear_candidates(candidates);
+                  m_num_available_points -= num_invalid;
```

Comment:
Can't you move that block into a function and use it everywhere?

Reply:
This is a good idea.

Reply:
Done.